CMakePresets.json

```diff
@@ -0,0 +1,28 @@
+{
+    "version": 3,
+    "configurePresets": [
+        {
+            "name": "macOS base",
+            "hidden": true,
+            "binaryDir": "${sourceDir}/build/",
+            "generator": "Xcode",
+            "cacheVariables": {
+                "CMAKE_PREFIX_PATH": "/usr/local/opt/qt@5"
```

Comment:
This path isn't standard on mac, I don't think? Definitely different on my system. I wonder if we can insert the output of `brew --prefix qt@5` here?

Reply:
Hm, the output of `brew --prefix qt@5` would be better, but it won't always be correct.
Is it still possible to configure/overwrite this on the user end? The prefix path would be different when e.g. building with an official Qt install.

Reply:
Presets are fully optional to use - they are not intended to cover every kind of configuration as this would be a duplication of the manual configuration via cli. But I agree, it would be the best way to cover as much variants as possible as long as it does not make the setup harder - sensible defaults which can be overwritten explicitly.

On the topic of if the output of a shell command could be used, I think this is not possible. At least I haven't found it in the [docs](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html).
But I think it could be possible via a cmake script which gets included via a build which can be set via a preset. The script could be something like 

```cmake
execute_process(
  COMMAND brew --prefix qt@5
  OUTPUT_VARIABLE QT5_BREW_PREFIX
  OUTPUT_STRIP_TRAILING_WHITESPACE
)
```

Maybe this would be a good script as most likely 95% of the macOS user base use brew to obtain qt5 and this would allow those users to not have to state the qt path explicitly anymore.